scipy/_lib/_util.py

```diff
@@ -153,6 +154,14 @@ def prod(iterable):
     return product
 
 
+def float_factorial(n: int) -> float:
+    """Compute the factorial and return as a float
+
+    Returns infinity when result is too large for a double
+    """
+    return float(math.factorial(n)) if n < 171 else np.inf
+
```

Comment:
Maybe also useful in
https://github.com/scipy/scipy/blob/f7f31b4574c9665afd87068fb466f78a7bc37dd6/scipy/interpolate/polyint.py#L307

Reply:
Had a grep around and found a few other `factorial` users as well. Not sure how feasible it would be to trigger an exception from those but it shouldn't hurt to be careful.

Reply:
thanks :)